File: CSharp/Contests2025/ABC407/E.cs
class E
{
	static void Main() => Console.WriteLine(string.Join("\n", new int[int.Parse(Console.ReadLine())].Select(_ => Solve())));
	static object Solve()
	{
		var n = int.Parse(Console.ReadLine());
		var a = Array.ConvertAll(new bool[2 * n], _ => long.Parse(Console.ReadLine()));

		var r = a[0];
		var q = new PriorityQueue<long, long>();

		for (int i = 1; i < n; i++)
		{
			var v = a[2 * i - 1];
			q.Enqueue(v, -v);
			v = a[2 * i];
			q.Enqueue(v, -v);
			r += q.Dequeue();
		}
		return r;
	}
}

File: Java/Contests2025/ABC407/src/E.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.PriorityQueue;

public class E {
	static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

	static long[] readL() throws IOException {
		return Arrays.stream(in.readLine().split(" ")).mapToLong(s -> Long.parseLong(s)).toArray();
	}

	static long[] readL(int n) throws IOException {
		var r = new long[n];
		for (int i = 0; i < n; i++)
			r[i] = Long.parseLong(in.readLine());
		return r;
	}

	public static void main(String[] args) throws IOException {
		var t = Integer.parseInt(in.readLine());
		var r = new String[t];
		for (int i = 0; i < t; i++)
			r[i] = solve().toString();
		System.out.println(String.join("\n", r));
	}

	static Object solve() throws IOException {
		var n = Integer.parseInt(in.readLine());
		var a = readL(2 * n);

		var r = a[0];
		var q = new PriorityQueue<Long>();

		for (int i = 1; i < n; i++) {
			q.offer(-a[2 * i - 1]);
			q.offer(-a[2 * i]);
			r += -q.remove();
		}
		return r;
	}
}
